frb/scripts/transient_crossmatching.py

```diff
@@ -0,0 +1,432 @@
+    the Mahalanobis distance from the FRB center.
+    
+    Parameters:
+    frbcenter (Astropy SkyCoord): The central position of the FRB.
+    cov_matrix (array-like): The covariance matrix representing the uncertainties in the FRB's position.
+    semi_major (float): The semi-major axis of the Gaussian ellipse in degrees.
+    transient_name (str): The name of the transient source.
+    transient_position (Astropy SkyCoord): Transient position.
+    levels: List of confidence levels for the contours (default: [0.68, 0.95, 0.99]).
+    """
+    
+    eigvals, eigvecs = np.linalg.eigh(cov_matrix)
+    order = eigvals.argsort()[::-1]
+    eigvals = eigvals[order]
+    eigvecs = eigvecs[:, order]
+    
+    width, height = 2 * np.sqrt(eigvals)
+    theta = np.degrees(np.arctan2(*eigvecs[:, 0][::-1]))
+    threshold = chi2.ppf(0.9999, df=2)
+    
+    # 2 arcmins size, quite arbitrary so can be changed
+    buffer = 2 * u.arcsec
+    if transient_position is not None:
+        size = (frbcenter.separation(transient_position)+buffer).to(u.arcmin)
+    else:
+        size = 10*u.arcmin #semi_major*60*30
+
+    fig = plt.figure(figsize=(6, 6))
+    ax = plt.axes(
+    projection='astro zoom',
+    center=frbcenter,
+    radius=size)
+
+    # Plot FRB center
+    ax.plot(frbcenter.ra.deg, frbcenter.dec.deg, 
+        transform=ax.get_transform('world'), color='black', marker='x', 
+        markersize=5)
+    
+    cmap = matplotlib.colormaps.get_cmap('viridis')
+
+    for i, level in enumerate(levels):
+        chi_square_val = chi2.ppf(level, 2)
+        color = cmap(i / len(levels))
+        ellipse = Ellipse(
+            xy=(frbcenter.ra.deg, frbcenter.dec.deg),
+            width=width * np.sqrt(chi_square_val),
+            height=height * np.sqrt(chi_square_val),
+            angle=theta,
+            edgecolor=color,
+            fc='None',
+            lw=2,
+            label=f'{int(level*100)}%',
+            transform=ax.get_transform('world')
+        )
+        ax.add_patch(ellipse)
+    
+    if transient_position is not None:
+        ax.plot(transient_position.ra.deg, transient_position.dec.deg, 
+                transform=ax.get_transform('world'), 
+                color='#fb5607', marker='o', label=transient_name)
+        
+        
+        md = mahalanobis_distance([transient_position.ra.deg, transient_position.dec.deg],
+                                  [frbcenter.ra.deg, frbcenter.dec.deg], cov_matrix)
+        pt = percentile(md)
+        ax.annotate(f'{(1-pt)*100:.3f}%', xy=[transient_position.ra.deg, transient_position.dec.deg],
+                    xytext=(12, 12), textcoords='offset points')     
+        
+    ax.set_xlabel('RA (J2000)', fontsize=14)
+    ax.set_ylabel('DEC (J2000)', fontsize=14)
+    ax.legend(fontsize=10)
+    ax.grid(False)
+    plt.tight_layout()
+    #plt.savefig(f'{transient_name}_gaussian_map.png')
+
+
+def main(filename, name, ra, dec, theta, a, b, radius, single_obj=False):
```

Comment:
are we going to make this a proper script?

If so, we need to add another file.

Reply:
let's discuss